internal/agent/event_sender.go

```diff
@@ -193,6 +195,8 @@ func (sender *metricsIngestSender) QueueEvent(event sample.Event, key entity.Key
 
 	select {
 	case sender.eventQueue <- queuedEvent:
+		metric := instrumentation.NewGauge("agent.eventQueueSize", float64(len(sender.eventQueue)))
```

Comment:
Would be nice to report capacity as well.

Calculating queue size here could be too frequent and cause performance issues.

Reply:
> Would be nice to report capacity as well.
 👍  We need to work on a list of "interesting" values to report

>Calculating queue size here could be too frequent and cause performance issues.
I think that we definitely should take the performance impact into account, but it needs to be a trade-off. In this case I don't expect to be as channels have a counter in the internal structure but I might be wrong.